### ISSUE TITLE
Terminix crashes when using F12 (session view) and PageDown/Up

### ISSUE THREAD

**Issue by sogalpunx** [gnunn1/terminix#684]
Hi,

After a fresh install on openSUSE (v1.4.0 via devel:languages:D repo), I was trying Terminix.
When I use F12 to open session side-bar and then PageUp or PageDown keys, Terminix crashes and dumps core:

`recvmsg(3, {msg_name(0)=NULL, msg_iov(1)=[{"#\203\2459\26\0\0\0\2\0\3\0k\374\"\1u\0\0\0\372\0\0\0\10\0\240\3\0\0\0\0"..., 4096}], msg_controllen=0, msg_flags=0}, 0) = 120
recvmsg(3, 0x7ffca8176be0, 0)           = -1 EAGAIN (Resource temporarily unavailable)
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0xfffffffffffffec0} ---
+++ killed by SIGSEGV (core dumped) +++`

**Comment by gnunn1** [comment]
I cannot reproduce on master, please try the last release which is 1.4.2 and see if it is still happening.

**Comment by gnunn1** [comment]
@sogalpunx Have you had a chance to try 1.4.2?

**Comment by sogalpunx** [comment]
Hi,
No chance at all.
On another machine, I've installed Terminix 1.4.2 via _OneClickInstall_ from _https://software.opensuse.org/package/terminix_. Same problem.
Then I've cloned the source and compile it with success, same problem.

In both cases, I did not change any default settings in Terminix.
I've just launched it, click on "+" or "Ctrl + Maj + t" to open a new session, then F12 to show the session side bar.
At this point, session 2 is selected in side bar. I can press PageUp to select session 1, no problem, but if I press PageDown to select back session 2, crash appends.

Actually, crashes only appends with PageDown.
I don't know I that might help you, but here's the xev output for my PageDown:
```
KeyPress event, serial 33, synthetic NO, window 0x2400001,
    root 0x26a, subw 0x0, time 1979127, (1480,734), root:(1578,892),
    state 0x10, keycode 117 (keysym 0xff56, Next), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyRelease event, serial 36, synthetic NO, window 0x2400001,
    root 0x26a, subw 0x0, time 1979231, (1480,734), root:(1578,892),
    state 0x10, keycode 117 (keysym 0xff56, Next), same_screen YES,
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False
```
I can use "Ctrl + PageUp" and "Ctrl + PageDown" to switch between session without problem.
Hope this helps, let me know if I can provide further information.

**Comment by gnunn1** [comment]
I still can't replicate the problem on master, it works fine for me. Can anyone else replicate this on 1.4.2?

**Comment by gnunn1** [comment]
@sogalpunx  The other option if you are willing is that I can compile a version from git that includes debug information, you could then use gdb to get a backtrace.

**Comment by sogalpunx** [comment]
> The other option if you are willing is that I can compile a version from git that includes debug information, you could then use gdb to get a backtrace.

Yes sure. I'll try Terminix in a virtual machine to see if I can reproduce it and I'll report the problem on an openSUSE forum, to see if somebody else can reproduce it as well.

**Comment by gnunn1** [comment]
I have built a version with debug enabled, so give it a whirl and see how it goes.

http://gexperts.com/files/terminix.zip

The issue might be related to different versions of GTK as well, I'm on the latest 3.22 version on Arch.

**Comment by super7ramp** [comment]
Hello,

> The issue might be related to different versions of GTK as well, I'm on the latest 3.22 version on Arch.

Indeed it's a regression of the GTK 3.20.x branch ([bgo#707703](https://bugzilla.gnome.org/show_bug.cgi?id=770703)). Both openSUSE Leap and SLE 12 use GTK 3.20.9.

[Fix](https://build.opensuse.org/request/show/451774) is coming.

**Comment by gnunn1** [comment]
Thanks @super7ramp, closing issue as upstream to resolve.

**Comment by sogalpunx** [comment]
Indeed, using a patched version of GTK 3.20 solved the problem.
Thanks to both of you.
